src/omotes_simulator_core/entities/network_controller.py

```diff
@@ -244,4 +277,8 @@ def _set_producers_based_on_priority(self, time: datetime.datetime) -> dict:
                     if source.priority == priority:
                         producers[source.id][PROPERTY_HEAT_DEMAND] = source.power
                 total_demand -= total_supply
+        if set_pressure:
```

Comment:
Why do you need to set this outside the loop? This could result in a system where not all assets are producing, but the assset that is not producing could still define the pressure?

Reply:
It can happen that no source is set to set the pressure then the system cannot be solved. Then we need to set the pressure of on of the source. It is not a problem if they are set to zero. They will then just define pressures in the system with zero flow.